pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4847,44 +4846,27 @@ cpdef to_offset(freq, bint is_period=False):
 
             tups = zip(split[0::4], split[1::4], split[2::4])
             for n, (sep, stride, name) in enumerate(tups):
-                if not is_period and name.upper() in c_OFFSET_DEPR_FREQSTR:
-                    warnings.warn(
-                        f"\'{name}\' is deprecated and will be removed "
-                        f"in a future version, please use "
-                        f"\'{c_OFFSET_DEPR_FREQSTR.get(name.upper())}\' instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
+                if not is_period and name.upper() in c_OFFSET_REMOVED_FREQSTR:
+                    raise ValueError(
+                        f"\'{name}\' is no longer supported for offsets."
                     )
-                    name = c_OFFSET_DEPR_FREQSTR[name.upper()]
                 if (not is_period and
                         name != name.upper() and
                         name.lower() not in {"s", "ms", "us", "ns"} and
                         name.upper().split("-")[0].endswith(("S", "E"))):
-                    warnings.warn(
-                        f"\'{name}\' is deprecated and will be removed "
-                        f"in a future version, please use "
-                        f"\'{name.upper()}\' instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
+                    raise ValueError(
+                        f"\'{name}\' is no longer supported for offsets."
                     )
-                    name = name.upper()
-                if is_period and name.upper() in c_REVERSE_OFFSET_DEPR_FREQSTR:
-                    if name.upper().startswith("Y"):
-                        raise ValueError(
-                            f"for Period, please use \'Y{name.upper()[2:]}\' "
-                            f"instead of \'{name}\'"
-                        )
+                if is_period and name.upper().split("-")[0].endswith(("E")):
                     if (name.upper().startswith("B") or
                             name.upper().startswith("S") or
                             name.upper().startswith("C")):
                         raise ValueError(INVALID_FREQ_ERR_MSG.format(name))
                     else:
                         raise ValueError(
-                            f"for Period, please use "
-                            f"\'{c_REVERSE_OFFSET_DEPR_FREQSTR.get(name.upper())}\' "
-                            f"instead of \'{name}\'"
+                            f"{name} is not supported as period frequency"
                         )
-                elif is_period and name.upper() in c_OFFSET_DEPR_FREQSTR:
+                elif is_period and name.upper() in c_OFFSET_REMOVED_FREQSTR:
```

Comment:
in this branch I'm seeing something like: if it's removed, then warn that it's deprecated

Is this right?

Do we need `c_OFFSET_REMOVED_FREQSTR` at all, or is it possible to just let it raise?

Reply:
agreed, it looks strange. It's my mistake, I will correct it.

I am unsure, maybe we can keep `c_OFFSET_REMOVED_FREQSTR` at least for a while. I think it would be good to explain when we raise why this frequency is invalid. Because it's possible to use this frequency in other cases, it wouldn't be entirely correct to say that it's `"Invalid frequency"`.

Reply:
Ok sure, that's a good point!

If we're going to keep it around then, maybe we can tell users which to use instead? At least for offsets?

Reply:
yes, I agree. I corrected the error message. Now we advise users which frequency for offsets to use instead of invalid one. I think CI failures are unrelated to my changes.